scripts/common.py

```diff
@@ -659,6 +659,10 @@ def wait_for_port_to_be_in_use(port_number):
         python_utils.PRINT(
             'Failed to start server on port %s, exiting ...' %
             port_number)
+        python_utils.PRINT(
+            'This may be because you do not have enough available '
+            'memory. Try quitting applications. We also recommend '
+            'using a machine with at least 8 GB RAM.')
```

Comment:
Should we write the recommended spec on the wiki page and add the link here? (Wiki page is easily editable and we won't have to change the codebase in the future if needed!)

Reply:
Yes, that's a great idea! The troubleshooting page also has more detailed workaround steps, so I've added a link to it

Reply:
Sorry for not being clear on my first comment, I was thinking of not mentioning any recommendation/suggestion in the code i.e, redirect users to the wiki page for recommendation/suggestions (As they are expected to get change in future as needed).  Something around:
```
'This may be because the system is out of memory. Please check the troubleshooting wiki page for help: https://github.com/oppia/oppia/wiki/Troubleshooting'

```

Reply:
I think that making new contributors go through the troubleshooting wiki page is an unnecessary obstacle to their contribution to Oppia. We want their onboarding process to be as easy as possible, so we should try and give them actionable feedback as quickly as possible.

If we need to change this message in the future, that would be as easy as opening another PR. Our guidance will probably change at most a few times a year, so it makes sense for us to go through a little extra work opening PRs in order to make the contribution experience easier.

What do you think? Note that I'll defer to you for an final decision on this since it's a dev workflow question.

Reply:
In general, we expect contributors to be familiar with wiki pages and reach wiki pages for any quick help/suggestion and we even ask the support team to always reply with a wiki link for a solution so that contributors know where to find answers to all general setup/process question. I think this falls under the same workflow!

Also, we have realized that we always miss updating the files whenever we are changing the wiki page (or vice-versa) (the reason being we don't track the wiki page and some files to be in sync.)

@BenHenning Do you have any suggestions on whether we should put the recommendation in the script or on the wiki page? [Asking this to you as you had some thoughts on "how to communicate the out-of-resource issue with contributors" during our meeting]

Reply:
@DubeySandeep I think you can make a call on this bearing in mind the multiple principles at play: (a) normalizing use of the wiki, (b) the concern about keeping code in sync with the wiki, (c) the importance of new contributors not needing to jump through hoops and get actionable feedback as quickly as possible.

One possible approach that might address all of this is to direct them to a specific bookmark on the wiki. If you point them to the Troubleshooting page, that is a lot of searching that may or may not get them anywhere.

Reply:
> One possible approach that might address all of this is to direct them to a specific bookmark on the wiki. If you point them to the Troubleshooting page, that is a lot of searching that may or may not get them anywhere.

This approach sounds good to me, but I'll still defer to you @DubeySandeep for the final decision

Reply:
I like the idea of linking directly to a page since it reduces hoops to jump through, but I also agree that links should take them directly to the thing they need rather than a generic page (hence less hoops to jump through)

Reply:
I've updated the troubleshooting doc to use headers instead of bullets so that we get direct links to each problem. The script now displays a link to the relevant troubleshooting item

Reply:
Thanks for your suggestions, @BenHenning & @seanlip I was expecting to have a link to a specific section on the wiki page.

@U8NWXD Thanks for adding the link! :)